app/controllers/api/configuration_script_sources_controller.rb

```diff
@@ -16,20 +16,15 @@ def delete_resource_main_action(type, config_script_src, _data)
       {:task_id => config_script_src.delete_in_provider_queue}
     end
 
-    def create_resource(_type, _id, data)
-      validate_attrs(data)
+    def create_resource(type, _id, data)
+      # TODO: convert configuration script api to (also?) support standard `ems_id` key
       manager_id = parse_id(data['manager_resource'], :providers)
-      raise 'Must specify a valid manager_resource href or id' unless manager_id
-      manager = resource_search(manager_id, :providers)
+      create_ems_resource(type, {'ems_id' => manager_id, 'name' => data['name']}) do |manager, klass|
+        # TODO: introduce supports for configuration scripts create
+        raise "ConfigurationScriptSource cannot be added to #{model_ident(manager, :provider)}" unless klass.respond_to?(:create_in_provider_queue)
 
-      type = "#{manager.type}::ConfigurationScriptSource"
-      klass = ConfigurationScriptSource.descendant_get(type)
-      raise "ConfigurationScriptSource cannot be added to #{manager_ident(manager)}" unless klass.respond_to?(:create_in_provider_queue)
-
-      task_id = klass.create_in_provider_queue(manager.id, data.except('manager_resource').deep_symbolize_keys)
-      action_result(true, "Creating ConfigurationScriptSource for #{manager_ident(manager)}", :task_id => task_id)
-    rescue => err
-      action_result(false, err.to_s)
+        {:task_id => klass.create_in_provider_queue(manager_id, data.except('manager_resource').deep_symbolize_keys)}
```

Comment:
Could you do `date.delete('manager_resource')` above if you don't want to pass that option through?